bugsnag-android-performance/src/test/java/com/bugsnag/android/performance/internal/AutoInstrumentationCacheTest.kt

```diff
@@ -0,0 +1,138 @@
+package com.bugsnag.android.performance.internal
+
+import android.app.Activity
+import android.app.Application
+import android.content.Context
+import android.content.pm.PackageInfo
+import android.content.pm.PackageManager
+import com.bugsnag.android.performance.AutoInstrumentationCache
+import com.bugsnag.android.performance.DoNotEndAppStart
+import com.bugsnag.android.performance.DoNotInstrument
+import com.bugsnag.android.performance.PerformanceConfiguration
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+
+internal class AutoInstrumentationCacheTest {
+    private val autoInstrumentationCache = AutoInstrumentationCache()
+
+    @DoNotInstrument
+    class DoNotInstrumentActivity : Activity()
+
+    @DoNotEndAppStart
+    class DoNotEndAppStartActivity : Activity()
+
+    @DoNotInstrument
+    @DoNotEndAppStart
+    class AnnotatedActivity : Activity()
+
+    class NoAnnotatedActivity : Activity()
+
+    private val perfConfig =
+        PerformanceConfiguration(mockedContext(), ImmutableConfigTest.TEST_API_KEY)
+
+    @Test
+    fun testDoNotInstrumentActivities() {
+        val activities = hashSetOf<Class<Any>>()
+        activities.add(DoNotInstrumentActivity::class.java as Class<Any>)
+        activities.add(DoNotEndAppStartActivity::class.java as Class<Any>)
+        perfConfig.doNotAutoInstrument = activities
```

Comment:
Until we can pass the set of non-annotated classes into the `AutoInstrumentationTest` this test isn't really doing anything that the other annotation-specific tests are covering.